### PR TITLE
Add the rest of the pre-existing metrics

### LINKED ISSUE #93: Add pairwise distance to Metrics
I think in evaluation of regression task, pairwise distance, especially norm-2 distance, as in `torch.nn.functional.pairwise_distance` is at least as frequently used as MSE, which is actually mostly used as loss rather than evaluation metrics. Therefore, I was wondering if it is worthy of being added to Metrics package as a commonly used metrics.

### PR DESCRIPTION
**Pull request by jasonkriss**
Resolves #93

### CODE REVIEW

File: ignite/metrics/root_mean_squared_error.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from __future__ import division
+import math
+
+import torch
+
+from .metric import Metric
+from ignite.exceptions import NotComputableError
+
+
+class RootMeanSquaredError(Metric):
```
Comment (alykhantejani) [inline ignite/metrics/root_mean_squared_error.py]: could this not inherit from MeanSquaredError and just override the `compute` function to be:
```python
def compute(self):
    mse = super(self, RootMeanSquaredError).compute()
    return math.sqrt(mse)
```

Comment (jasonkriss) [reply ignite/metrics/root_mean_squared_error.py]: Definitely. One other option is to just have MeanSquaredError but have it take a `root` flag in the constructor. I think I prefer the more explicit approach of having both MeanSquaredError and RootMeanSquaredError though. What do you think?

Comment (jasonkriss) [reply ignite/metrics/root_mean_squared_error.py]: @alykhantejani I went ahead and made your suggested change. I can definitely update it though if you wanna go a different route.

**Comment by jasonkriss** [comment]
@alykhantejani I just added MeanPairwiseDistance to this PR.

File: ignite/metrics/root_mean_squared_error.py (hunk 1 above)
Comment (alykhantejani) [reply ignite/metrics/root_mean_squared_error.py]: 👍 I also like the explicit route
